### DISCUSSION TITLE
Web Start (.jnlp)

### DISCUSSION THREAD

**Discussion by cohe** [microsoft/openjdk#25] [Q&A] [answered]
Once upon a time there was **WebStart** function part of Java JRE/JDK package as well as all JavaFX components.

After using AdoptOpenJDK + IcedTea for web starts + customized compiled JavaFX binaries
there is a new hope in this galaxy to have easier access to up-to-date Java packages containing a JRE capable of Web start and JavaFX applications.

**Q:** Will the Microsoft Java build already contain a web start feature?
(or is it required/compatible to use it with OpenWebStart)

Best regards and thank you in advance,
cohe

**Comment by brunoborges** [comment, marked as answer]
Hi @cohe,

Currently, we do not have plans to extend the Microsoft Build of OpenJDK with other features, including OpenWebStart, that are not part of the OpenJDK project.

For this scenario, we do recommend the use of `jlink` and `jpackage` to produce OS native package installers with the Java runtime bundled within. Companies like [GluonHQ](https://www.gluonhq.com) provide great support for this.

If customer demand shows the need for such a feature, we may reconsider in the future.

Thanks,
Bruno
